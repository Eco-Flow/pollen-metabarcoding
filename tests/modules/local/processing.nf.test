nextflow_process {

    name "Test Process PROCESSING"
    script "modules/local/processing.nf"
    process "PROCESSING"

    test("Should run without failures") {

        when {
            params {
                clean = false
            }
            process {
                """
                input[0] = Channel.of( [ [ "id":"ERR2537816", "single_end": false ], file("${projectDir}/data/processing-input/ERR2537816.tsv") ] ) 
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            with(process.out) {
              assert fasta.size() == 1
              assert fasta[0].size() == 2
              assert fasta[0][0].size() == 2
              assert path(fasta[0].get(1)).md5 == '58f98a56c0746e50cf57e944d80c3dc0'
            }

        }

    }

}
