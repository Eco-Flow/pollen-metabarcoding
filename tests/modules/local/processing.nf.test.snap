{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "ERR2537816",
                            "single_end": false
                        },
                        "ERR2537816.classified.tsv:md5,58f98a56c0746e50cf57e944d80c3dc0"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "id": "ERR2537816",
                            "single_end": false
                        },
                        "ERR2537816.classified.tsv:md5,58f98a56c0746e50cf57e944d80c3dc0"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T11:34:38.230732"
    }
}